### PR TITLE
Added a way to apply a function to a list of datasets without needing globals etc. to retain the function's return values.

### PR DESCRIPTION
**Pull request by DCichon**
Changed 'loop_over_datasets' into 'function_results_datasets', which allows to apply a function to datasets and, at the same time, to receive the function's return value for each event. Created a new 'loop_over_datasets' with the same functionality and parameters as the old one to maintain backward compatibility.

### CODE REVIEW

File: hax/paxroot.py [hunk 1]
```diff
@@ -39,11 +39,15 @@ def open_pax_rootfile(filename):
 class StopEventLoop(Exception):
     pass
 
-
-def loop_over_datasets(datasets_names, event_function=lambda event: None, branch_selection='basic'):
-    """Execute event_function(event) over all events in the dataset(s)
-    Does not return anything: you have to keep track of results yourself (global vars, function attrs, classes, ...)
-    branch selection: can be None (all branches are read), 'basic' (hax.config['basic_branches'] are read), or a list of branches to read.
+def function_results_datasets(datasets_names, event_function=lambda event, **kwargs: None,
```
Comment (tunnell) [inline hax/paxroot.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: hax/paxroot.py [hunk 2]
```diff
@@ -78,12 +82,22 @@ def loop_over_datasets(datasets_names, event_function=lambda event: None, branch
...
                 event = t.events
-                event_function(event)
+                yield event_function(event, **kwargs)
         except StopEventLoop:
             rootfile.Close()
         except Exception as e:
             rootfile.Close()
             raise e
 
+def loop_over_datasets(datasets_names, event_function=lambda event: None,
```
Comment (tunnell) [inline hax/paxroot.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: hax/paxroot.py [hunk 3]
```diff
@@ -78,12 +82,22 @@ def loop_over_datasets(datasets_names, event_function=lambda event: None, branch
...
+                       branch_selection='basic'):
+    """Execute event_function(event) over all events in the dataset(s)
+    Does not return anything: you have to keep track of results yourself (global vars, function attrs, classes, ...)
+    branch selection: can be None (all branches are read), 'basic' (hax.config['basic_branches'] are read), or a list of branches to read.
```
Comment (tunnell) [inline hax/paxroot.py]: __[flake8]__
* [E501] line too long (138 > 120 characters)

**Comment by tunnell** [comment]
@DCichon It looks like a useful change and is nicely implemented.  I have no objections.  @JelleAalbers I assume you're also fine with it?  Will merge after our pax release.

**Comment by JelleAalbers** [comment]
Hi Dominic, this is the right solution! Thanks for contributing.

**Pull request merged**